### PR TITLE
[rush] Relax the rush.json version check for rush-lib

### PR DESCRIPTION
**Pull request by pgonzal**
Future versions are now accepted as long as the major/minor parts match.

### CODE REVIEW

File: apps/rush-lib/src/data/RushConfiguration.ts [hunk 1]
```diff
@@ -173,14 +173,29 @@ export default class RushConfiguration {
...
-        throw new Error(`Unable to load ${rushJsonBaseName} because its RushVersion is`
-          + ` ${rushConfigurationJson.rushVersion}, whereas @microsoft/rush-lib is version ${Rush.version}.`
-          + ` Consider upgrading the library.`);
-      } else if (semver.lt(expectedRushVersion, MINIMUM_SUPPORTED_RUSH_JSON_VERSION)) {
+      // Make sure the requested version isn't too old
+      if (semver.lt(expectedRushVersion, MINIMUM_SUPPORTED_RUSH_JSON_VERSION)) {
         throw new Error(`${rushJsonBaseName} is version ${expectedRushVersion}, which is too old for this tool. ` +
           `The minimum supported version is ${MINIMUM_SUPPORTED_RUSH_JSON_VERSION}.`);
       }
+console.log('whoopsie: ' + Rush.version + ' vs ' + expectedRushVersion);
```
Comment (iclanton) [inline apps/rush-lib/src/data/RushConfiguration.ts]: >whoopsie [](start = 13, length = 8)

?

Comment (pgonzal) [reply apps/rush-lib/src/data/RushConfiguration.ts]: oops, that was some debug code

Comment (pgonzal) [reply apps/rush-lib/src/data/RushConfiguration.ts]: oops, that was some debug code

---
In reply to: [183514399](https://github.com/Microsoft/web-build-tools/pull/618#discussion_r183514399) [](ancestors = 183514399)

File: apps/rush-lib/src/data/RushConfiguration.ts [hunk 2]
```diff
@@ -173,14 +173,29 @@ export default class RushConfiguration {
 
     // If the version is missing or malformed, fall through and let the schema handle it.
     if (expectedRushVersion && semver.valid(expectedRushVersion)) {
-      if (semver.lt(Rush.version, expectedRushVersion)) {
-        throw new Error(`Unable to load ${rushJsonBaseName} because its RushVersion is`
-          + ` ${rushConfigurationJson.rushVersion}, whereas @microsoft/rush-lib is version ${Rush.version}.`
-          + ` Consider upgrading the library.`);
-      } else if (semver.lt(expectedRushVersion, MINIMUM_SUPPORTED_RUSH_JSON_VERSION)) {
+      // Make sure the requested version isn't too old
+      if (semver.lt(expectedRushVersion, MINIMUM_SUPPORTED_RUSH_JSON_VERSION)) {
```
Comment (iclanton) [inline apps/rush-lib/src/data/RushConfiguration.ts]: We were talking about introducing a schema/configuration version field in rush.json. I think that would be cleaner. Then we very explicitly increment the version if we make a change.

Comment (pgonzal) [reply apps/rush-lib/src/data/RushConfiguration.ts]: I'll come chat about this

---
In reply to: [183514953](https://github.com/Microsoft/web-build-tools/pull/618#discussion_r183514953) [](ancestors = 183514953)

Comment (pgonzal) [reply apps/rush-lib/src/data/RushConfiguration.ts]: We discussed this and Ian was okay with the approach

---
In reply to: [183515290](https://github.com/Microsoft/web-build-tools/pull/618#discussion_r183515290) [](ancestors = 183515290,183514953)

**Pull request merged**